### PR TITLE
Add "intuition_solver" indirection in intuition to allow for redefinion

### LINKED ISSUE #7725: There should be `Set Intuition Solver` like `Set Firstorder Solver`
#### Version

8.8.0

#### Description of the problem

It is very unfortunate that `intuition` means `intuition auto with *`.  It seems like adding this option would be a step towards making the default behavior here easier to change, as I suggest in #7724.

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #7725

### CODE REVIEW

**Comment by JasonGross** [comment]
> Not sure if I should try changing to `auto` in this PR (ie fix #4949)

I think you should maybe save that for a separate PR.  I imagine that change will be invasive and require many overlays, and I don't want this change to get bogged down if that one takes too long to fix.

**Review by JasonGross — APPROVED** [review]

**Comment by SkySkimmer** [comment]
Merge on then

**Comment by JasonGross** [comment]
Do we need a review from @coq/stdlib-maintainers ?

File: theories/Init/Tauto.v
Comment (Alizter) [inline theories/Init/Tauto.v:107]: Why is idtac needed here?

Comment (SkySkimmer) [reply theories/Init/Tauto.v:107]: Without it, if the solver starts with a `match goal` (as in the added test) it gets executed too soon.

Comment (JasonGross) [reply theories/Init/Tauto.v:107]: Because Ltac semantics are insane.  The test-suite example will fail without the idtac, because `match`s get evaluated at the call site rather than the use site, but we want the solver tactic run at the leaves.

**Comment by SkySkimmer** [comment]
@JasonGross I don't think we need to wait.

**Comment by JasonGross** [comment]
@coqbot merge now

**Pull request merged**
